File: python/generador.py
'''
Generador de tablas de multiplicar
'''

import utilidades as util

def main() -> None:
  numero: int = 0
  inicio: int = 0
  fin: int = 0
  
  util.encabezado('Generador de tablas de multiplicar', 32)
  numero = util.stringToNumber('Ingresa tu numero: ', int)
  
  util.encabezado(f'tabla de multiplicar del {numero}', 32)
  generarTabla(numero)
  
  util.encabezado('Generador de rango de tablas', 32)
  inicio = util.stringToNumber('Ingresa tu numero de inicio: ', int)
  fin = util.stringToNumber('Ingresa tu numero de final: ', int)
  
  util.encabezado(f'tabla de multiplicar del {inicio} al {fin}', 32)
  generarRango(inicio, fin)
  
def generarRango(inicio: int, fin: int) -> None:
  # nroTablas: int = abs(inicio - fin) + 1, abs es para indicar un valor absoluto no negativo
  
  for j in range(inicio, fin + 1):
    util.linea(32)
    generarTabla(j)
  util.linea(32)
  
def generarTabla(numero: int) -> None:
  for i in range(1, 11):
    print(f'{numero} x {i} = {producto(numero, i)}')
producto = lambda a,b : a * b

main()

File: Js/src/generador.js
/*
Generador de tablas de multiplicar
*/

import {encabezado, line, readLineNumber} from './utilidades.js';

const main = () => {
  //VARIABLES
  let numero = 0
  let inicio = 0
  let final = 0

  const producto = (a, b) => a * b

  const generarTabla = (numero = 0) => {
    for (let i = 0; i < 11; i++) {
      console.log(`${numero} x ${i} = ${producto(numero, i)}`)
    }
  }

   const generarRango = (inicio, final) =>{
    for (inicio; inicio <= final; inicio++) {
      for (let i = 0; i <= 11; i++) {
        console.log(`${inicio} x ${i} = ${producto(inicio, i)}`)
      }
      line(32)
    }
  }

  encabezado('Generador de tablas de multiplicar', 48)
  numero = readLineNumber('ingresa un numero')
  encabezado(`Tabla del ${numero}`, 48)
  generarTabla(numero);

  encabezado('Has un rango de tablas de multiplicar', 48)
  inicio = readLineNumber('Ingresa tu numero de inicio')
  final = readLineNumber('Ingresa tu numero de final')
  encabezado(`tablas de multiplicar de ${inicio} al ${final}`, 48)
  generarRango(inicio, final)

}

main()